packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -258,4 +318,339 @@ describe('Select', () => {
+
+        const filterModeTestData = [
+            {
+                filter: FilterMode.none,
+                name: 'none'
+            },
+            {
+                filter: FilterMode.standard,
+                name: 'standard'
+            }
+        ];
+        filterModeTestData.forEach(testData => {
+            it(`pressing <Enter> opens dropdown when filterMode = ${testData.name}`, () => {
+                element.filterMode = testData.filter;
+                pageObject.pressEnterKey();
+                expect(element.open).toBeTrue();
+            });
+
+            it(`pressing <Space> opens dropdown when filterMode = ${testData.name}`, async () => {
+                element.filterMode = testData.filter;
+                await pageObject.pressSpaceKey();
+                expect(element.open).toBeTrue();
+            });
+
+            it(`after pressing <Esc> to close dropdown, <Enter> will re-open dropdown when filterMode = ${testData.name}`, async () => {
+                element.filterMode = testData.filter;
+                await pageObject.clickSelect();
+                pageObject.pressEscapeKey();
+                expect(element.open).toBeFalse();
+                pageObject.pressEnterKey();
+                expect(element.open).toBeTrue();
+            });
+
+            it(`after closing dropdown by pressing <Esc>, activeElement is Select element when filterMode = ${testData.name}`, async () => {
+                element.filterMode = testData.filter;
+                await pageObject.clickSelect();
+                pageObject.pressEscapeKey();
+                expect(document.activeElement).toBe(element);
+            });
+
+            it(`after closing dropdown by committing a value with <Enter>, activeElement is Select element when filterMode = ${testData.name}`, async () => {
+                element.filterMode = testData.filter;
+                await pageObject.clickSelect();
+                pageObject.pressArrowDownKey();
+                pageObject.pressEnterKey();
+                expect(document.activeElement).toBe(element);
+            });
+        });
+    });
+
+    describe('filtering', () => {
+        let element: Select;
+        let connect: () => Promise<void>;
+        let disconnect: () => Promise<void>;
+        let pageObject: SelectPageObject;
+
+        beforeEach(async () => {
+            ({ element, connect, disconnect } = await setup());
+            element.style.width = '200px';
+            element.filterMode = FilterMode.standard;
+            await connect();
+            pageObject = new SelectPageObject(element);
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        it('opening dropdown puts focus in filter input', async () => {
+            await clickAndWaitForOpen(element);
+            expect(document.activeElement!.shadowRoot!.activeElement).toBe(
+                element.filterInputElement!
+            );
+        });
```

Comment:
@rajsite @jattasNI, after chatting with @mollykreis we realized that the proper way to test for focus (as we've discovered before, and I just forgot) is to test the `activeElement` of the `shadowRoot` of the current `activeElement` if the focus was given to something in the shadow DOM.